src/motion/Motor.cpp

```diff
@@ -40,14 +43,25 @@ Motor::Motor(uint8_t _enablePin, uint8_t _dirPin, uint8_t _clockPin, bool _invDi
 void Motor::update() {
 	// compute speed according to current position
 	int32_t speed;
-	if(position < accelerationDistance) // acceleration phase
+
+	if(recalPhase && position < goalPosition) { // recalibration phase
+		speed = recalSpeed;
+	} else if(position < accelerationDistance) { // acceleration phase
 		speed = minSpeed + ((accelerationCoeff*position & 0xFFFFFC00) >> MOTOR_ACC_COEFF_FP);
-	else if(position < goalPosition - accelerationDistance) // cruise phase (constant speed)
+	} else if(position < goalPosition - accelerationDistance) { // cruise phase (constant speed)
```

Comment:
I don't understand this condition. Shouldn't it be:
```suggestion
	} else if((position > accelerationDistance) && (position < goalPosition)) { // cruise phase (constant speed)
```